containers/Dockerfile.al2

```diff
@@ -1,90 +1,108 @@
-FROM amazonlinux as builder
+# Copyright 2020-2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+FROM amazonlinux:2 as builder
+
+RUN yum install -y \
+	cmake3 \
+	gcc \
+	git \
+	tar \
+	make \
+	gcc-c++ \
+	go \
+	ninja-build \
+	doxygen
 
-RUN yum upgrade -y
-RUN amazon-linux-extras enable epel
-RUN yum clean -y metadata && yum install -y epel-release
-RUN yum install -y cmake3 gcc git tar make
 RUN curl https://sh.rustup.rs -sSf | sh -s -- -y --profile minimal --default-toolchain 1.60
 
-RUN yum install -y gcc-c++
-RUN yum install -y go
-RUN yum install -y ninja-build
-
 # We keep the build artifacts in the -build directory
 WORKDIR /tmp/crt-builder
 
-RUN git clone -b v1.0.2 https://github.com/awslabs/aws-lc.git aws-lc #
-RUN cmake3 -DCMAKE_PREFIX_PATH=/usr -DCMAKE_INSTALL_PREFIX=/usr -GNinja -S aws-lc -B aws-lc/build .
+RUN git clone -b v1.0.2 https://github.com/awslabs/aws-lc.git aws-lc
+RUN cmake3 -DCMAKE_PREFIX_PATH=/usr -DCMAKE_INSTALL_PREFIX=/usr -GNinja -DBUILD_TESTING=0 -S aws-lc -B aws-lc/build .
 RUN go env -w GOPROXY=direct
-RUN cmake3 --build aws-lc/build --target install
+RUN cmake3 --build aws-lc/build --parallel $(nproc) --target install
 
 RUN git clone -b v1.3.20 https://github.com/aws/s2n-tls.git
-RUN cmake3 -DCMAKE_PREFIX_PATH=/usr -DCMAKE_INSTALL_PREFIX=/usr -S s2n-tls -B s2n-tls/build
-RUN cmake3 --build s2n-tls/build --target install
+RUN cmake3 -DCMAKE_PREFIX_PATH=/usr -DBUILD_TESTING=0 -DCMAKE_INSTALL_PREFIX=/usr -S s2n-tls -B s2n-tls/build
```

Comment:
nit: What about combining each group of builds in one RUN? 

```
RUN group1-item1 && \
         group1-item2
         
RUN group2-item1 && \
         group2-item2
```

Reply:
Yes, but that only decreases the overall container size slightly. We would like to keep in more command parts since in case we need to debug to know which `RUN` failed.